### ISSUE TITLE
Issue with var parameter + inheritance in the VM

### ISSUE THREAD

**Issue by GULPF** [nim-lang/Nim#11861]
### Example
```nim
type
  Foo* = object of RootObj
    field*: int

  Bar* = object of Foo

proc mutate(l: var Foo) =
  l.field.inc

proc test(): int =
  var l = Bar()
  l.mutate()
  result = l.field

doAssert test() == 1 # Ok
static:
  doAssert test() == 1 # Fails
```

### Expected Output
Both asserts should pass.

**Comment by mratsim** [comment]
The issue is with the l.mutate() statement.

Similar to #11637 it seems like `mutate` actually mutates a copy of the input and the passed parameter is left alone, this can be seen by adding 2 doAssert statements.

```Nim
type
  Foo* = object of RootObj
    field*: int

  Bar* = object of Foo

proc mutate(l: var Foo) =
  l.field.inc
  doAssert l.field == 1

proc test(): int =
  var l = Bar()
  l.mutate()
  doAssert l.field == 1 # fails here
  result = l.field

doAssert test() == 1 # Ok
static:
  doAssert test() == 1 # Fails
```

**Comment by Clyybber** [comment]
The issue is in downConv ccgexpr.nim.
As I rely on this working correctly in the new destructor spec, I have included a fix in #11284
